--- intltool-0.50.2/intltool.m4.bak	Fri Jun  7 04:14:50 2013
+++ intltool-0.50.2/intltool.m4	Fri Jun  7 04:15:52 2013
@@ -116,20 +116,12 @@
 _IT_SUBST(INTLTOOL_SERVICE_RULE)
 _IT_SUBST(INTLTOOL_POLICY_RULE)
 
-# Check the gettext tools to make sure they are GNU
 AC_PATH_PROG(XGETTEXT, xgettext)
-AC_PATH_PROG(MSGMERGE, msgmerge)
 AC_PATH_PROG(MSGFMT, msgfmt)
 AC_PATH_PROG(GMSGFMT, gmsgfmt, $MSGFMT)
-if test -z "$XGETTEXT" -o -z "$MSGMERGE" -o -z "$MSGFMT"; then
-    AC_MSG_ERROR([GNU gettext tools not found; required for intltool])
+if test -z "$XGETTEXT" -o -z "$MSGFMT"; then
+    AC_MSG_ERROR([gettext tools not found; required for intltool])
 fi
-xgversion="`$XGETTEXT --version|grep '(GNU ' 2> /dev/null`"
-mmversion="`$MSGMERGE --version|grep '(GNU ' 2> /dev/null`"
-mfversion="`$MSGFMT --version|grep '(GNU ' 2> /dev/null`"
-if test -z "$xgversion" -o -z "$mmversion" -o -z "$mfversion"; then
-    AC_MSG_ERROR([GNU gettext tools not found; required for intltool])
-fi
 
 AC_PATH_PROG(INTLTOOL_PERL, perl)
 if test -z "$INTLTOOL_PERL"; then
