--- cups-filters-1.0.30/filter/pdftopdf/pptypes.cc.bak	2013-05-28 06:44:26.798301274 -0700
+++ cups-filters-1.0.30/filter/pdftopdf/pptypes.cc	2013-05-28 06:47:53.252660787 -0700
@@ -163,10 +163,17 @@
 
 void PageRect::set(const PageRect &rhs) // {{{
 {
+#ifdef __sun
+  if (!isnan(rhs.top)) top=rhs.top;
+  if (!isnan(rhs.left)) left=rhs.left;
+  if (!isnan(rhs.right)) right=rhs.right;
+  if (!isnan(rhs.bottom)) bottom=rhs.bottom;
+#else
   if (!std::isnan(rhs.top)) top=rhs.top;
   if (!std::isnan(rhs.left)) left=rhs.left;
   if (!std::isnan(rhs.right)) right=rhs.right;
   if (!std::isnan(rhs.bottom)) bottom=rhs.bottom;
+#endif
 }
 // }}}
 
--- cups-filters-1.0.30/filter/pdftopdf/pptypes.h.bak	2013-05-28 06:42:56.746362911 -0700
+++ cups-filters-1.0.30/filter/pdftopdf/pptypes.h	2013-05-28 06:43:53.079377596 -0700
@@ -3,6 +3,10 @@
 
 #include <cmath> // NAN
 
+#ifndef NAN
+#define NAN 0.0/0.0
+#endif
+
 // namespace PPTypes {}   TODO?
 
 enum Axis { X, Y };
