diff --git a/usr/src/Makefile.master b/usr/src/Makefile.master
index 27de79e..01386ca 100644
--- a/usr/src/Makefile.master
+++ b/usr/src/Makefile.master
@@ -22,6 +22,7 @@
 #
 # Copyright (c) 1989, 2010, Oracle and/or its affiliates. All rights reserved.
 # Copyright (c) 2012 by Delphix. All rights reserved.
+# Copyright (c) 2013 Andrew Stormont.  All rights reserved.
 #
 
 #
@@ -748,7 +749,7 @@ NATIVE_MACH=	$(MACH:amd64=i386)
 #
 SPRO_ROOT=		$(BUILD_TOOLS)/SUNWspro
 SPRO_VROOT=		$(SPRO_ROOT)/SS12
-GNU_ROOT=		$(SFW_ROOT)
+GNU_ROOT=		/usr/gnu
 
 # Till SS12u1 formally becomes the NV CBE, LINT is hard
 # coded to be picked up from the $SPRO_ROOT/sunstudio12.1/
diff --git a/usr/src/psm/stand/bootblks/ufs/i386/Makefile b/usr/src/psm/stand/bootblks/ufs/i386/Makefile
index f41d526..b135976 100644
--- a/usr/src/psm/stand/bootblks/ufs/i386/Makefile
+++ b/usr/src/psm/stand/bootblks/ufs/i386/Makefile
@@ -23,6 +23,8 @@
 # Copyright 2005 Sun Microsystems, Inc.  All rights reserved.
 # Use is subject to license terms.
 #
+# Copyright (c) 2013 Andrew Stormont.  All rights reserved.
+#
 # psm/stand/bootblks/ufs/i386/Makefile
 #
 .KEEP_STATE:
@@ -36,10 +38,10 @@ CC		= $(GCC_ROOT)/bin/gcc
 ASFLAGS		= -B$(GCC_ROOT)/bin/ -fno-builtin -nostdinc
 CPPFLAGS	=
 
-LD		= $(GNU_ROOT)/bin/gld
+LD		= $(GNU_ROOT)/bin/ld
 LDFLAGS		= -nostdlib -N -Ttext 600
 
-OBJCOPY		= $(GNU_ROOT)/bin/gobjcopy
+OBJCOPY		= $(GNU_ROOT)/bin/objcopy
 
 INSTALL_DIR	= $(USR)/lib/fs/ufs
 INSTALL_TARGETS	= $(PROGS:%=$(INSTALL_DIR)/%)
diff --git a/usr/src/grub/grub-0.97/Makefile.solaris.defs b/usr/src/grub/grub-0.97/Makefile.solaris.defs
index 9a9455e..84ab651 100644
--- a/usr/src/grub/grub-0.97/Makefile.solaris.defs
+++ b/usr/src/grub/grub-0.97/Makefile.solaris.defs
@@ -2,6 +2,8 @@
 # Copyright 2005 Sun Microsystems, Inc.  All rights reserved.
 # Use is subject to license terms.
 #
+# Copyright (c) 2013 Andrew Stormont.  All rights reserved.
+#
 include $(SRC)/Makefile.master
 
 .KEEP_STATE:
@@ -68,7 +70,7 @@ CPPFLAGS	= $(DEFS) $(INCLUDES) $(WARNFLAGS) \
 		  $(ENVCPPFLAGS3:-I%=-isystem %) \
 		  $(t.ENVCPPFLAGS4:I,%=-nostdinc -isystem %)
 
-CCLD		= $(GNU_ROOT)/bin/gld
+CCLD		= $(GNU_ROOT)/bin/ld
 LDFLAGS		= $(BASE_LDFLAGS)
 LINKFLAGS	= -g
 LINK		= $(CCLD) $(LINKFLAGS) $(LDFLAGS)
@@ -78,7 +80,7 @@ OPTFLAGS	= -falign-jumps=1 -falign-loops=1 -falign-functions=1
 #WARNFLAGS	= -Wall -Wmissing-prototypes -Wunused -Wshadow \
 #		  -Wpointer-arith -Wundef
 
-OBJCOPY		= $(GNU_ROOT)/bin/gobjcopy
+OBJCOPY		= $(GNU_ROOT)/bin/objcopy
 
 AR		= /usr/ccs/bin/ar
 RANLIB		= :
