diff --git a/usr/src/Makefile.master b/usr/src/Makefile.master
index 27de79e..bc34435 100644
--- a/usr/src/Makefile.master
+++ b/usr/src/Makefile.master
@@ -748,7 +748,6 @@ NATIVE_MACH=	$(MACH:amd64=i386)
 #
 SPRO_ROOT=		$(BUILD_TOOLS)/SUNWspro
 SPRO_VROOT=		$(SPRO_ROOT)/SS12
-GNU_ROOT=		$(SFW_ROOT)
 
 # Till SS12u1 formally becomes the NV CBE, LINT is hard
 # coded to be picked up from the $SPRO_ROOT/sunstudio12.1/
diff --git a/usr/src/grub/grub-0.97/Makefile.solaris.defs b/usr/src/grub/grub-0.97/Makefile.solaris.defs
index 9a9455e..ccca243 100644
--- a/usr/src/grub/grub-0.97/Makefile.solaris.defs
+++ b/usr/src/grub/grub-0.97/Makefile.solaris.defs
@@ -1,6 +1,7 @@
 #
 # Copyright 2005 Sun Microsystems, Inc.  All rights reserved.
 # Use is subject to license terms.
+# Copyright (c) 2013 Andrew Stormont.  All rights reserved.
 #
 include $(SRC)/Makefile.master
 
@@ -68,7 +69,7 @@ CPPFLAGS	= $(DEFS) $(INCLUDES) $(WARNFLAGS) \
 		  $(ENVCPPFLAGS3:-I%=-isystem %) \
 		  $(t.ENVCPPFLAGS4:I,%=-nostdinc -isystem %)
 
-CCLD		= $(GNU_ROOT)/bin/gld
+CCLD		= /usr/bin/gld
 LDFLAGS		= $(BASE_LDFLAGS)
 LINKFLAGS	= -g
 LINK		= $(CCLD) $(LINKFLAGS) $(LDFLAGS)
@@ -78,7 +79,7 @@ OPTFLAGS	= -falign-jumps=1 -falign-loops=1 -falign-functions=1
 #WARNFLAGS	= -Wall -Wmissing-prototypes -Wunused -Wshadow \
 #		  -Wpointer-arith -Wundef
 
-OBJCOPY		= $(GNU_ROOT)/bin/gobjcopy
+OBJCOPY		= /usr/bin/gobjcopy
 
 AR		= /usr/ccs/bin/ar
 RANLIB		= :
diff --git a/usr/src/psm/stand/bootblks/ufs/i386/Makefile b/usr/src/psm/stand/bootblks/ufs/i386/Makefile
index f41d526..61dc749 100644
--- a/usr/src/psm/stand/bootblks/ufs/i386/Makefile
+++ b/usr/src/psm/stand/bootblks/ufs/i386/Makefile
@@ -22,6 +22,7 @@
 #
 # Copyright 2005 Sun Microsystems, Inc.  All rights reserved.
 # Use is subject to license terms.
+# Copyright (c) 2013 Andrew Stormont.  All rights reserved.
 #
 # psm/stand/bootblks/ufs/i386/Makefile
 #
@@ -36,10 +37,10 @@ CC		= $(GCC_ROOT)/bin/gcc
 ASFLAGS		= -B$(GCC_ROOT)/bin/ -fno-builtin -nostdinc
 CPPFLAGS	=
 
-LD		= $(GNU_ROOT)/bin/gld
+LD		= /usr/bin/gld
 LDFLAGS		= -nostdlib -N -Ttext 600
 
-OBJCOPY		= $(GNU_ROOT)/bin/gobjcopy
+OBJCOPY		= /usr/bin/gobjcopy
 
 INSTALL_DIR	= $(USR)/lib/fs/ufs
 INSTALL_TARGETS	= $(PROGS:%=$(INSTALL_DIR)/%)
diff --git a/usr/src/tools/aw/Makefile b/usr/src/tools/aw/Makefile
index af2bf8f..7125c03 100644
--- a/usr/src/tools/aw/Makefile
+++ b/usr/src/tools/aw/Makefile
@@ -22,6 +22,7 @@
 #
 # Copyright 2004 Sun Microsystems, Inc.  All rights reserved.
 # Use is subject to license terms.
+# Copyright (c) 2013 Andrew Stormont.  All rights reserved.
 #
 
 PROG	= aw
@@ -33,8 +34,8 @@ CERRWARN += -_gcc=-Wno-uninitialized
 
 LINTFLAGS += -ux -Xa -errchk=locfmtchk,parentheses
 
-CPPFLAGS +=	-DDEFAULT_AS_DIR='"$(GNU_ROOT)/bin"'
-CPPFLAGS +=	-DDEFAULT_AS64_DIR='"$(GNU_ROOT)/bin"'
+CPPFLAGS +=	-DDEFAULT_AS_DIR='"/usr/bin"'
+CPPFLAGS +=	-DDEFAULT_AS64_DIR='"/usr/bin"'
 CPPFLAGS +=	-DDEFAULT_M4_DIR='"/usr/ccs/bin"'
 CPPFLAGS +=	-DDEFAULT_M4LIB_DIR='"/usr/ccs/lib"'
 CPPFLAGS +=	-DDEFAULT_CPP_DIR='"/usr/ccs/lib"'
diff --git a/usr/src/cmd/nsadmin/dot-profile.sh b/usr/src/cmd/nsadmin/dot-profile.sh
index 33a79c9..3bdc7ec 100644
--- a/usr/src/cmd/nsadmin/dot-profile.sh
+++ b/usr/src/cmd/nsadmin/dot-profile.sh
@@ -1,10 +1,10 @@
 #
-# Simple profile places /usr/gnu/bin at front,
-# adds /usr/X11/bin, /usr/sbin and /sbin to the end.
+# Simple profile places /usr/bin at front,
+# adds /usr/sbin and /sbin to the end.
 #
 # Use less(1) as the default pager for the man(1) command.
 #
-export PATH=/usr/gnu/bin:/usr/bin:/usr/sbin:/sbin
+export PATH=/usr/bin:/usr/sbin:/sbin
 export PAGER="/usr/bin/less -ins"
 
 #
