diff --git a/usr/src/cmd/egrep/egrep.y b/usr/src/cmd/egrep/egrep.y
index 95e071e..9d616f8 100644
--- a/usr/src/cmd/egrep/egrep.y
+++ b/usr/src/cmd/egrep/egrep.y
@@ -26,6 +26,8 @@
  * Use is subject to license terms.
  */
 
+/* Copyright (c) 2013 Andrew Stormont.  All rights reserved. */
+
 /*	Copyright (c) 1984, 1986, 1987, 1988, 1989 AT&T	*/
 /*	  All Rights Reserved  	*/
 
@@ -114,6 +116,7 @@ int	hflag;
 int	iflag;
 int	lflag;
 int	nflag;
+int	qflag;
 int	sflag;
 int	vflag;
 int	nfile;
@@ -644,7 +647,7 @@ follow(int v)
 	}
 }
 
-#define USAGE "[ -bchilnsv ] [ -e exp ] [ -f file ] [ strings ] [ file ] ..." 
+#define USAGE "[ -bchilnqsv ] [ -e exp ] [ -f file ] [ strings ] [ file ] ..." 
 
 int
 main(int argc, char **argv)
@@ -660,7 +663,7 @@ main(int argc, char **argv)
 #endif
 	(void) textdomain(TEXT_DOMAIN);
 
-	while((c = getopt(argc, argv, "ybcie:f:hlnvs")) != -1)
+	while((c = getopt(argc, argv, "ybcie:f:hlnqvs")) != -1)
 		switch(c) {
 
 		case 'b':
@@ -703,6 +706,10 @@ main(int argc, char **argv)
 			nflag++;
 			continue;
 
+		case 'q':
+			qflag++;
+			continue;
+
 		case 's':
 			sflag++;
 			continue;
@@ -904,7 +911,10 @@ execute(char *file)
 		}
 		if(succ) {
 			nsucc = 1;
-			if (cflag) tln++;
+			if (qflag)
+				return;
+			else if (cflag)
+				tln++;
 			else if (sflag)
 				;	/* ugh */
 			else if (lflag) {
