diff --git a/usr/src/cmd/ssh/libssh/Makefile.com b/usr/src/cmd/ssh/libssh/Makefile.com
index 569afe7..15212bb 100644
--- a/usr/src/cmd/ssh/libssh/Makefile.com
+++ b/usr/src/cmd/ssh/libssh/Makefile.com
@@ -20,6 +20,7 @@
 #
 # Copyright 2009 Sun Microsystems, Inc.  All rights reserved.
 # Use is subject to license terms.
+# Copyright (c) 2013 Andrew Stormont.  All rights reserved.
 #
 
 LIBRARY =	libssh.a
@@ -91,6 +92,9 @@ SRCS =		$(OBJECTS:%.o=../common/%.c)
 
 LIBS =		$(LIBRARY) $(LINTLIB)
 
+# Handle case where pkcs11 engine support is not wanted
+$(ENABLE_PKCS11_ENGINE)CPPFLAGS += -DENABLE_PKCS11_ENGINE
+
 # Define LDLIBS conditionally for lintcheck, rather than in general, since
 # we're building an archive library which itself links to nothing, we just
 # want lint to know about these libraries.
diff --git a/usr/src/cmd/ssh/libssh/common/engine.c b/usr/src/cmd/ssh/libssh/common/engine.c
index 0541c65..4c90300 100644
--- a/usr/src/cmd/ssh/libssh/common/engine.c
+++ b/usr/src/cmd/ssh/libssh/common/engine.c
@@ -22,6 +22,7 @@
  * Copyright 2008 Sun Microsystems, Inc.  All rights reserved.
  * Use is subject to license terms.
  */
+/* Copyright (c) 2013 Andrew Stormont.  All rights reserved. */
 
 #include "includes.h"
 #include "log.h"
@@ -38,6 +39,7 @@ pkcs11_engine_load(int use_engine)
 {
 	ENGINE *e = NULL;
 
+#ifdef ENABLE_PKCS11_ENGINE
 	debug("use_engine is '%s'", use_engine == 1 ? "yes" : "no");
 	if (use_engine == 0)
 		return (NULL);
@@ -84,6 +86,9 @@ pkcs11_engine_load(int use_engine)
 	}
 
 	debug("%s engine initialization complete", PKCS11_ENGINE);
+#else
+	debug("pkcs11 engine support not compiled in");
+#endif
 	return (e);
 }
 
