diff --git a/usr/src/cmd/find/find.c b/usr/src/cmd/find/find.c
index a8754e8..29f19af 100644
--- a/usr/src/cmd/find/find.c
+++ b/usr/src/cmd/find/find.c
@@ -21,6 +21,7 @@
 /*
  * Copyright (c) 1988, 2010, Oracle and/or its affiliates. All rights reserved.
  * Copyright 2012 Nexenta Systems, Inc. All rights reserved.
+ * Copyright (c) 2013 Andrew Stormont.  All rights reserved.
  */
 
 
@@ -83,9 +84,9 @@ enum Command
 {
 	PRINT,
 	ACL, AMIN, AND, ATIME, CMIN, CPIO, CSIZE, CTIME, DEPTH, EXEC, F_GROUP,
-	F_GROUPACL, F_USER, F_USERACL, FOLLOW, FSTYPE, INAME, INUM, IREGEX,
+	F_GROUPACL, F_USER, F_USERACL, FOLLOW, FSTYPE, INAME, INUM, IPATH, IREGEX,
 	LINKS, LOCAL, LPAREN, LS, MAXDEPTH, MINDEPTH, MMIN, MOUNT, MTIME, NAME,
-	NCPIO, NEWER, NOGRP, NOT, NOUSER, OK, OR, PERM, PRINT0, PRUNE, REGEX,
+	NCPIO, NEWER, NOGRP, NOT, NOUSER, OK, OR, PATH, PERM, PRINT0, PRUNE, REGEX,
 	RPAREN, SIZE, TYPE, VARARGS, XATTR
 };
 
@@ -125,6 +126,7 @@ static struct Args commands[] =
 	"-groupacl",	F_GROUPACL,	Num,
 	"-iname",	INAME,		Str,
 	"-inum",	INUM,		Num,
+	"-ipath",	IPATH,		Str,
 	"-iregex",	IREGEX,		Str,
 	"-links",	LINKS,		Num,
 	"-local",	LOCAL,		Unary,
@@ -143,6 +145,7 @@ static struct Args commands[] =
 	"-o",		OR,		Op,
 	"-ok",		OK,		Exec,
 	"-or",		OR,		Op,
+	"-path",	PATH,		Str,
 	"-perm",	PERM,		Num,
 	"-print",	PRINT,		Unary,
 	"-print0",	PRINT0,		Unary,
@@ -621,6 +624,8 @@ int *actionp;
 
 		case NAME:
 		case INAME:
+		case PATH:
+		case IPATH:
 			np->first.cp = b;
 			break;
 		case REGEX:
@@ -993,16 +998,18 @@ struct FTW *state;
 			break;
 
 		case NAME:
-		case INAME: {
-			char *name1;
-			int fnmflags = (np->action == INAME) ?
+		case INAME:
+		case PATH:
+		case IPATH: {
+			char *path;
+			int fnmflags = (np->action == INAME || np->action == IPATH) ?
 			    FNM_IGNORECASE : 0;
 
 			/*
 			 * basename(3c) may modify name, so
 			 * we need to pass another string
 			 */
-			if ((name1 = strdup(name)) == NULL) {
+			if ((path = strdup(name)) == NULL) {
 				(void) fprintf(stderr,
 				    gettext("%s: cannot strdup() %s: %s\n"),
 				    cmdname, name, strerror(errno));
@@ -1018,8 +1025,9 @@ struct FTW *state;
 #ifndef XPG4
 			fnmflags |= FNM_PERIOD;
 #endif
-			val = !fnmatch(np->first.cp, basename(name1), fnmflags);
-			free(name1);
+			val = !fnmatch(np->first.cp, (np->action == PATH || np->action == IPATH) ?
+			    path : basename(path), fnmflags);
+			free(path);
 			break;
 		}
 
