diff --git a/usr/src/cmd/fm/eversholt/common/esclex.c b/usr/src/cmd/fm/eversholt/common/esclex.c
index 2c3bf73..35eb1fb 100644
--- a/usr/src/cmd/fm/eversholt/common/esclex.c
+++ b/usr/src/cmd/fm/eversholt/common/esclex.c
@@ -21,6 +21,7 @@
 /*
  * Copyright 2008 Sun Microsystems, Inc.  All rights reserved.
  * Use is subject to license terms.
+ * Copyright (c) 2013 Andrew Stormont.  All rights reserved.
  *
  * esclex.c -- lexer for esc
  *
@@ -73,7 +74,7 @@ static const char *Fileopened;
 static FILE *Fp;
 static int Line;
 static const char *File;
-static const char *Cpp = "/usr/bin/cpp";
+static const char *Cpp = "/usr/lib/cpp";
 #ifdef	ESC
 static const char *Cppargs;
 static const char *Cppstdargs = "-undef -Y.";
