diff --git a/usr/src/Makefile.master b/usr/src/Makefile.master
index bc34435..bdf3b2d 100644
--- a/usr/src/Makefile.master
+++ b/usr/src/Makefile.master
@@ -22,6 +22,7 @@
 #
 # Copyright (c) 1989, 2010, Oracle and/or its affiliates. All rights reserved.
 # Copyright (c) 2012 by Delphix. All rights reserved.
+# Copyright (c) 2013 Andrew Stormont.  All rights reserved.
 #
 
 #
@@ -572,7 +573,7 @@ CLOBBERFILES += $(POFILE) $(POFILES)
 COMPILE.cpp= $(CC) -E -C $(CFLAGS) $(CPPFLAGS)
 XGETTEXT= /usr/bin/xgettext
 XGETFLAGS= -c TRANSLATION_NOTE
-GNUXGETTEXT= /usr/gnu/bin/xgettext
+GNUXGETTEXT= /usr/bin/gxgettext
 GNUXGETFLAGS= --add-comments=TRANSLATION_NOTE --keyword=_ \
 	--strict --no-location --omit-header
 BUILD.po= $(XGETTEXT) $(XGETFLAGS) -d $(<F) $<.i ;\
diff --git a/usr/src/cmd/ssh/Makefile b/usr/src/cmd/ssh/Makefile
index 8a53794..a86ebe9 100644
--- a/usr/src/cmd/ssh/Makefile
+++ b/usr/src/cmd/ssh/Makefile
@@ -22,6 +22,7 @@
 #
 # Copyright 2009 Sun Microsystems, Inc.  All rights reserved.
 # Use is subject to license terms.
+# Copyright (c) 2013 Andrew Stormont.  All rights reserved.
 #
 
 include		../Makefile.cmd
@@ -75,7 +76,7 @@ check: $(CHECKHDRS)
 _msg:
 	$(RM) $(POFILE)
 	$(TOUCH) $(POFILE)
-	$(MAKE) $(POFILE) XGETTEXT=/usr/bin/gxgettext
+	$(MAKE) $(POFILE) XGETTEXT=$(GNUXGETTEXT)
 	$(CP) $(POFILE) $(MSGFILE)
 	$(CP) $(MSGFILE) $(MSGDOMAIN)
 
