diff --git a/usr/src/lib/libkmf/plugins/kmf_openssl/common/openssl_spi.c b/usr/src/lib/libkmf/plugins/kmf_openssl/common/openssl_spi.c
index fc3da4b..12dcf55 100644
--- a/usr/src/lib/libkmf/plugins/kmf_openssl/common/openssl_spi.c
+++ b/usr/src/lib/libkmf/plugins/kmf_openssl/common/openssl_spi.c
@@ -7,6 +7,9 @@
  * Copyright (c) 2012, OmniTI Computer Consulting, Inc. All rights reserved.
  */
 /*
+ * Copyright (c) 2013 Andrew Stormont.  All rights reserved.
+ */
+/*
  * Written by Dr Stephen N Henson (shenson@bigfoot.com) for the OpenSSL
  * project 2000.
  */
@@ -1802,8 +1805,10 @@ OpenSSL_SignData(KMF_HANDLE_T handle, KMF_KEY_HANDLE *key,
 		int len;
 		if (AlgId == KMF_ALGID_MD5WithRSA)
 			md = EVP_md5();
+#ifndef OPENSSL_NO_MD2
 		else if (AlgId == KMF_ALGID_MD2WithRSA)
 			md = EVP_md2();
+#endif
 		else if (AlgId == KMF_ALGID_SHA1WithRSA)
 			md = EVP_sha1();
 		else if (AlgId == KMF_ALGID_SHA256WithRSA)
