diff --combined usr/src/lib/Makefile
index bae90c4,bae90c4..0000000
--- a/usr/src/lib/Makefile
+++ b/usr/src/lib/Makefile
@@@ -23,6 -23,6 +23,7 @@@
  # Copyright (c) 1989, 2010, Oracle and/or its affiliates. All rights reserved.
  # Copyright (c) 2012 by Delphix. All rights reserved.
  # Copyright (c) 2012, Joyent, Inc. All rights reserved.
++# Copyright (c) 2013 Andrew Stormont.  All rights reserved.
  
  include ../Makefile.master
  
@@@ -75,6 -75,6 +76,7 @@@ SUBDIRS += 
  	libresolv2		.WAIT	\
  	libw			.WAIT	\
  	libintl			.WAIT	\
++	libiconv		\
  	../cmd/sgs/librtld_db	\
  	libaio			\
  	libast			\
diff --combined usr/src/lib/libaio/Makefile
index 6cb7a4f,6cb7a4f..0000000
--- a/usr/src/lib/libaio/Makefile
+++ b/usr/src/lib/libaio/Makefile
@@@ -21,11 -21,11 +21,13 @@@
  #
  # Copyright 2009 Sun Microsystems, Inc.  All rights reserved.
  # Use is subject to license terms.
++# Copyright (c) 2013 Andrew Stormont.  All rights reserved.
  #
  
  include		../Makefile.lib
  
--SUBDIRS =	$(MACH) $(BUILD64) $(MACH64)
++SUBDIRS =	$(MACH)
++$(BUILD64)SUBDIRS += $(MACH64)
  
  all :=		TARGET= all
  clean :=	TARGET= clean
diff --git a/usr/src/lib/libiconv/Makefile b/usr/src/lib/libiconv/Makefile
new file mode 100644
index 0000000..d745bfe
--- /dev/null
+++ b/usr/src/lib/libiconv/Makefile
@@ -0,0 +1,45 @@
+#
+# CDDL HEADER START
+#
+# The contents of this file are subject to the terms of the
+# Common Development and Distribution License (the "License").
+# You may not use this file except in compliance with the License.
+#
+# You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
+# or http://www.opensolaris.org/os/licensing.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+#
+# When distributing Covered Code, include this CDDL HEADER in each
+# file and include the License file at usr/src/OPENSOLARIS.LICENSE.
+# If applicable, add the following below this CDDL HEADER, with the
+# fields enclosed by brackets "[]" replaced with your own identifying
+# information: Portions Copyright [yyyy] [name of copyright owner]
+#
+# CDDL HEADER END
+#
+# Copyright (c) 2013 Andrew Stormont.  All rights reserved.
+#
+
+include ../Makefile.lib
+
+SUBDIRS = $(MACH)
+$(BUILD64)SUBDIRS += $(MACH64)
+
+all	:= TARGET= all
+clean	:= TARGET= clean
+clobber	:= TARGET= clobber
+install	:= TARGET= install
+lint	:= TARGET= lint
+
+.KEEP_STATE:
+
+all clean clobber install lint: $(SUBDIRS)
+
+$(SUBDIRS): FRC
+	@cd $@; pwd; $(MAKE) $(TARGET)
+
+FRC:
+
+include $(SRC)/lib/Makefile.targ
+include $(SRC)/Makefile.msg.targ
diff --git a/usr/src/lib/libiconv/Makefile.com b/usr/src/lib/libiconv/Makefile.com
new file mode 100644
index 0000000..d024b60
--- /dev/null
+++ b/usr/src/lib/libiconv/Makefile.com
@@ -0,0 +1,38 @@
+#
+# CDDL HEADER START
+#
+# The contents of this file are subject to the terms of the
+# Common Development and Distribution License (the "License").
+# You may not use this file except in compliance with the License.
+#
+# You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
+# or http://www.opensolaris.org/os/licensing.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+#
+# When distributing Covered Code, include this CDDL HEADER in each
+# file and include the License file at usr/src/OPENSOLARIS.LICENSE.
+# If applicable, add the following below this CDDL HEADER, with the
+# fields enclosed by brackets "[]" replaced with your own identifying
+# information: Portions Copyright [yyyy] [name of copyright owner]
+#
+# CDDL HEADER END
+#
+# Copyright (c) 2013 Andrew Stormont. All rights reserved.
+#
+
+LIBRARY	=	libiconv.a
+VERS =		.1
+
+include ../../Makefile.lib
+
+# disable ctfmerge on this library
+CTFMERGE_LIB =
+
+.KEEP_STATE:
+
+all:
+
+lint:	lintcheck
+
+include ../../Makefile.targ
diff --git a/usr/src/lib/libiconv/amd64/Makefile b/usr/src/lib/libiconv/amd64/Makefile
new file mode 100644
index 0000000..c178c47
--- /dev/null
+++ b/usr/src/lib/libiconv/amd64/Makefile
@@ -0,0 +1,27 @@
+#
+# CDDL HEADER START
+#
+# The contents of this file are subject to the terms of the
+# Common Development and Distribution License, Version 1.0 only
+# (the "License").  You may not use this file except in compliance
+# with the License.
+#
+# You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
+# or http://www.opensolaris.org/os/licensing.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+#
+# When distributing Covered Code, include this CDDL HEADER in each
+# file and include the License file at usr/src/OPENSOLARIS.LICENSE.
+# If applicable, add the following below this CDDL HEADER, with the
+# fields enclosed by brackets "[]" replaced with your own identifying
+# information: Portions Copyright [yyyy] [name of copyright owner]
+#
+# CDDL HEADER END
+#
+# Copyright (c) 2013 Andrew Stormont.  All rights reserved.
+#
+
+include ../Makefile.com
+
+install: all $(ROOTLIBS) $(ROOTLINKS)
diff --git a/usr/src/lib/libiconv/i386/Makefile b/usr/src/lib/libiconv/i386/Makefile
new file mode 100644
index 0000000..c178c47
--- /dev/null
+++ b/usr/src/lib/libiconv/i386/Makefile
@@ -0,0 +1,27 @@
+#
+# CDDL HEADER START
+#
+# The contents of this file are subject to the terms of the
+# Common Development and Distribution License, Version 1.0 only
+# (the "License").  You may not use this file except in compliance
+# with the License.
+#
+# You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
+# or http://www.opensolaris.org/os/licensing.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+#
+# When distributing Covered Code, include this CDDL HEADER in each
+# file and include the License file at usr/src/OPENSOLARIS.LICENSE.
+# If applicable, add the following below this CDDL HEADER, with the
+# fields enclosed by brackets "[]" replaced with your own identifying
+# information: Portions Copyright [yyyy] [name of copyright owner]
+#
+# CDDL HEADER END
+#
+# Copyright (c) 2013 Andrew Stormont.  All rights reserved.
+#
+
+include ../Makefile.com
+
+install: all $(ROOTLIBS) $(ROOTLINKS)
diff --git a/usr/src/lib/libiconv/mapfile-vers b/usr/src/lib/libiconv/mapfile-vers
new file mode 100644
index 0000000..5bfd716
--- /dev/null
+++ b/usr/src/lib/libiconv/mapfile-vers
@@ -0,0 +1,66 @@
+#
+# CDDL HEADER START
+#
+# The contents of this file are subject to the terms of the
+# Common Development and Distribution License (the "License").
+# You may not use this file except in compliance with the License.
+#
+# You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
+# or http://www.opensolaris.org/os/licensing.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+#
+# When distributing Covered Code, include this CDDL HEADER in each
+# file and include the License file at usr/src/OPENSOLARIS.LICENSE.
+# If applicable, add the following below this CDDL HEADER, with the
+# fields enclosed by brackets "[]" replaced with your own identifying
+# information: Portions Copyright [yyyy] [name of copyright owner]
+#
+# CDDL HEADER END
+#
+#
+# Copyright (c) 2006, 2010, Oracle and/or its affiliates. All rights reserved.
+# Copyright (c) 2013 Andrew Stormont. All rights reserved.
+#
+
+$mapfile_version 2
+
+# Mnemonic conditional input identifiers:
+#
+# - amd64, i386, sparc32, sparcv9: Correspond to ISA subdirectories used to
+#	hold per-platform code. Note however that we use 'sparc32' instead of
+#	'sparc'. Since '_sparc' is predefined to apply to, all sparc platforms,
+#	naming the 32-bit version 'sparc' would be too likely to cause errors.
+#
+# -	lf64: Defined on platforms that offer the 32-bit largefile APIs
+#
+$if _ELF32
+$add lf64
+$endif
+$if _sparc && _ELF32
+$add sparc32
+$endif
+$if _sparc && _ELF64
+$add sparcv9
+$endif
+$if _x86 && _ELF32
+$add i386
+$endif
+$if _x86 && _ELF64
+$add amd64
+$endif
+
+SYMBOL_VERSION STORMOS_0.1 {
+    global:
+$if _ELF32
+	iconv { TYPE = FUNCTION; FILTER = /usr/lib/libc.so.1 };
+$elif sparcv9
+	iconv { TYPE = FUNCTION; FILTER = /usr/lib/sparcv9/libc.so.1 };
+$elif amd64
+	iconv { TYPE = FUNCTION; FILTER = /usr/lib/amd64/libc.so.1 };
+$else
+$error unknown platform
+$endif
+    local:
+	*;
+};
diff --git a/usr/src/lib/libiconv/sparc/Makefile b/usr/src/lib/libiconv/sparc/Makefile
new file mode 100644
index 0000000..c178c47
--- /dev/null
+++ b/usr/src/lib/libiconv/sparc/Makefile
@@ -0,0 +1,27 @@
+#
+# CDDL HEADER START
+#
+# The contents of this file are subject to the terms of the
+# Common Development and Distribution License, Version 1.0 only
+# (the "License").  You may not use this file except in compliance
+# with the License.
+#
+# You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
+# or http://www.opensolaris.org/os/licensing.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+#
+# When distributing Covered Code, include this CDDL HEADER in each
+# file and include the License file at usr/src/OPENSOLARIS.LICENSE.
+# If applicable, add the following below this CDDL HEADER, with the
+# fields enclosed by brackets "[]" replaced with your own identifying
+# information: Portions Copyright [yyyy] [name of copyright owner]
+#
+# CDDL HEADER END
+#
+# Copyright (c) 2013 Andrew Stormont.  All rights reserved.
+#
+
+include ../Makefile.com
+
+install: all $(ROOTLIBS) $(ROOTLINKS)
diff --git a/usr/src/lib/libiconv/sparcv9/Makefile b/usr/src/lib/libiconv/sparcv9/Makefile
new file mode 100644
index 0000000..c178c47
--- /dev/null
+++ b/usr/src/lib/libiconv/sparcv9/Makefile
@@ -0,0 +1,27 @@
+#
+# CDDL HEADER START
+#
+# The contents of this file are subject to the terms of the
+# Common Development and Distribution License, Version 1.0 only
+# (the "License").  You may not use this file except in compliance
+# with the License.
+#
+# You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
+# or http://www.opensolaris.org/os/licensing.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+#
+# When distributing Covered Code, include this CDDL HEADER in each
+# file and include the License file at usr/src/OPENSOLARIS.LICENSE.
+# If applicable, add the following below this CDDL HEADER, with the
+# fields enclosed by brackets "[]" replaced with your own identifying
+# information: Portions Copyright [yyyy] [name of copyright owner]
+#
+# CDDL HEADER END
+#
+# Copyright (c) 2013 Andrew Stormont.  All rights reserved.
+#
+
+include ../Makefile.com
+
+install: all $(ROOTLIBS) $(ROOTLINKS)
