diff --git a/usr/src/Targetdirs b/usr/src/Targetdirs
index 494231b..0321700 100644
--- a/usr/src/Targetdirs
+++ b/usr/src/Targetdirs
@@ -936,8 +936,8 @@ $(ROOT)/usr/lib/libgen.so.1:=		REALPATH=../../lib/libgen.so.1
 $(ROOT)/usr/lib/libgen.so:=		REALPATH=../../lib/libgen.so.1
 $(ROOT)/usr/lib/libinetutil.so.1:=	REALPATH=../../lib/libinetutil.so.1
 $(ROOT)/usr/lib/libinetutil.so:=	REALPATH=../../lib/libinetutil.so.1
-$(ROOT)/usr/lib/libintl.so.1:=		REALPATH=../../lib/libintl.so.1
-$(ROOT)/usr/lib/libintl.so:=		REALPATH=../../lib/libintl.so.1
+$(ROOT)/usr/lib/libsunw_intl.so.1:=	REALPATH=../../lib/libsunw_intl.so.1
+$(ROOT)/usr/lib/libsunw_intl.so:=	REALPATH=../../lib/libsunw_intl.so.1
 $(ROOT)/usr/lib/libkmf.so.1:=		REALPATH=../../lib/libkmf.so.1
 $(ROOT)/usr/lib/libkmf.so:=		REALPATH=../../lib/libkmf.so.1
 $(ROOT)/usr/lib/libkmfberder.so.1:=	REALPATH=../../lib/libkmfberder.so.1
@@ -1226,10 +1226,10 @@ $(ROOT)/usr/lib/$(MACH64)/libinetutil.so.1:= \
 	REALPATH=../../../lib/$(MACH64)/libinetutil.so.1
 $(ROOT)/usr/lib/$(MACH64)/libinetutil.so:= \
 	REALPATH=../../../lib/$(MACH64)/libinetutil.so.1
-$(ROOT)/usr/lib/$(MACH64)/libintl.so.1:= \
-	REALPATH=../../../lib/$(MACH64)/libintl.so.1
-$(ROOT)/usr/lib/$(MACH64)/libintl.so:= \
-	REALPATH=../../../lib/$(MACH64)/libintl.so.1
+$(ROOT)/usr/lib/$(MACH64)/libsunw_intl.so.1:= \
+	REALPATH=../../../lib/$(MACH64)/libsunw_intl.so.1
+$(ROOT)/usr/lib/$(MACH64)/libsunw_intl.so:= \
+	REALPATH=../../../lib/$(MACH64)/libsunw_intl.so.1
 $(ROOT)/usr/lib/$(MACH64)/libkstat.so.1:= \
 	REALPATH=../../../lib/$(MACH64)/libkstat.so.1
 $(ROOT)/usr/lib/$(MACH64)/libkstat.so:= \
@@ -1552,8 +1552,8 @@ SYM.USRLIB= \
 	/usr/lib/libgen.so.1 \
 	/usr/lib/libinetutil.so \
 	/usr/lib/libinetutil.so.1 \
-	/usr/lib/libintl.so \
-	/usr/lib/libintl.so.1 \
+	/usr/lib/libsunw_intl.so \
+	/usr/lib/libsunw_intl.so.1 \
 	/usr/lib/libkstat.so \
 	/usr/lib/libkstat.so.1 \
 	/usr/lib/liblddbg.so.4 \
@@ -1798,8 +1798,8 @@ SYM.USRLIB64= \
 	/usr/lib/$(MACH64)/libgen.so.1 \
 	/usr/lib/$(MACH64)/libinetutil.so \
 	/usr/lib/$(MACH64)/libinetutil.so.1 \
-	/usr/lib/$(MACH64)/libintl.so \
-	/usr/lib/$(MACH64)/libintl.so.1 \
+	/usr/lib/$(MACH64)/libsunw_intl.so \
+	/usr/lib/$(MACH64)/libsunw_intl.so.1 \
 	/usr/lib/$(MACH64)/libkstat.so \
 	/usr/lib/$(MACH64)/libkstat.so.1 \
 	/usr/lib/$(MACH64)/liblddbg.so.4 \
diff --git a/usr/src/lib/libintl/Makefile.com b/usr/src/lib/libintl/Makefile.com
index 78ac289..bb69d49 100644
--- a/usr/src/lib/libintl/Makefile.com
+++ b/usr/src/lib/libintl/Makefile.com
@@ -21,12 +21,15 @@
 #
 # Copyright 2009 Sun Microsystems, Inc.  All rights reserved.
 # Use is subject to license terms.
+# Copyright (c) 2013 Andrew Stormont.  All rights reserved.
 #
 
-LIBRARY =	libintl.a
+LIBRARY =	libsunw_intl.a
 VERS =		.1
 
 include		$(SRC)/lib/Makefile.rootfs
 
+LINTSRC=	$(LINTLIB:llib-lsunw_%.ln=llib-l%)
+
 LIBS +=		$(LINTLIB)
 DYNFLAGS +=	-F libc.so.1
