diff --git a/usr/src/cmd/rm/rm.c b/usr/src/cmd/rm/rm.c
index f3815b3..b25a2e2 100644
--- a/usr/src/cmd/rm/rm.c
+++ b/usr/src/cmd/rm/rm.c
@@ -75,7 +75,7 @@ static int confirm(FILE *, const char *, ...);
 static void memerror(void);
 static int checkdir(struct dlist *, struct dlist *);
 static int errcnt;
-static boolean_t silent, interactive, recursive, ontty;
+static boolean_t silent, interactive, recursive, verbose, ontty;
 
 static char *pathbuf;
 static size_t pathbuflen = MAXPATHLEN;
@@ -95,7 +95,7 @@ main(int argc, char **argv)
 #endif
 	(void) textdomain(TEXT_DOMAIN);
 
-	while ((c = getopt(argc, argv, "frRi")) != EOF)
+	while ((c = getopt(argc, argv, "frRiv")) != EOF)
 		switch (c) {
 		case 'f':
 			silent = B_TRUE;
@@ -113,6 +113,9 @@ main(int argc, char **argv)
 		case 'R':
 			recursive = B_TRUE;
 			break;
+		case 'v':
+			verbose = B_TRUE;
+			break;
 		case '?':
 			errflg = 1;
 			break;
@@ -554,7 +557,8 @@ unlinkit:
 #endif
 		}
 		errcnt++;
-	}
+	} else if (verbose)
+		printf(gettext("removed `%s'\n"), pathbuf);
 	return (0);
 }
 
