--- Python-2.6.8/Modules/socketmodule.c.bak	2013-04-22 10:39:08.025092979 -0700
+++ Python-2.6.8/Modules/socketmodule.c	2013-04-22 10:49:40.555781136 -0700
@@ -1096,7 +1096,7 @@
         }
 #endif
 
-#ifdef HAVE_NETPACKET_PACKET_H
+#if defined(HAVE_NETPACKET_PACKET_H) && defined(SIOCGIFNAME)
     case AF_PACKET:
     {
         struct sockaddr_ll *a = (struct sockaddr_ll *)addr;
@@ -1387,7 +1387,7 @@
     }
 #endif
 
-#ifdef HAVE_NETPACKET_PACKET_H
+#if defined(HAVE_NETPACKET_PACKET_H) && defined(SIOCGIFINDEX)
     case AF_PACKET:
     {
         struct sockaddr_ll* addr;
@@ -4607,9 +4607,13 @@
     PyModule_AddIntConstant(m, "PACKET_MULTICAST", PACKET_MULTICAST);
     PyModule_AddIntConstant(m, "PACKET_OTHERHOST", PACKET_OTHERHOST);
     PyModule_AddIntConstant(m, "PACKET_OUTGOING", PACKET_OUTGOING);
+#ifdef PACKET_LOOPBACK
     PyModule_AddIntConstant(m, "PACKET_LOOPBACK", PACKET_LOOPBACK);
+#endif
+#ifdef PACKET_FASTROUTE
     PyModule_AddIntConstant(m, "PACKET_FASTROUTE", PACKET_FASTROUTE);
 #endif
+#endif
 
 #ifdef HAVE_LINUX_TIPC_H
     PyModule_AddIntConstant(m, "AF_TIPC", AF_TIPC);
